### PR TITLE
Refactor Ownership Library for better UX

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- Swaps the need for declaring ownership in storage for a pre-defined `StorageKey` to store storage. This results in the following improvements in developer UX as the use of the library was previously very verbose.

Before:
```sway
storage {
     owner: Ownership = Ownership::initialized(Identity::Address(Address::from(ZERO_B256))),
}

fn foo() {
     let owner = storage.owner.owner();
     storage.owner.only_owner();
     storage.owner.renounce_ownership();
}
```
After:
```sway
fn foo() {
     let owner = _owner();
     only_owner();
     renounce_ownership();
}
```

The downside to this approach is the inability to set the owner during compile time and a constructor must be implemented.

### CODE REVIEW

**Review by Braqzen — COMMENTED** [review]
Before I review the changes themselves can you elaborate on the technical need for this change?

To me it seems like a subjective change in UX.
I think that having the ability to set data at compile time is important and useful. At the same time being able to delay that via a constructor may be useful as well.

Without deeper consideration at this stage I'll await your response and instead superficially propose the idea of having 2 ownership standards. One for configuration and one for construction. That has trade-offs as well but I don't think that's as much of an issue as providing one UX over the other.

**Comment by bitzoic** [comment]
> Before I review the changes themselves can you elaborate on the technical need for this change?
> 
> To me it seems like a subjective change in UX. I think that having the ability to set data at compile time is important and useful. At the same time being able to delay that via a constructor may be useful as well.
> 
> Without deeper consideration at this stage I'll await your response and instead superficially propose the idea of having 2 ownership standards. One for configuration and one for construction. That has trade-offs as well but I don't think that's as much of an issue as providing one UX over the other.

This is something I considered, but there are a few caveats to this. 
1. In the future we expect to have the ability to import storage from libraries, thus enabling the ability to set the owner at compile time
2. With most developers coming from Solidity, a constructor is something they are already accustomed to
3. If a developer chooses to set the owner at compile time, they can do so using just the [SRC-5 standard](https://github.com/FuelLabs/sway-standards/blob/master/examples/src_5/initialized_example/src/initialized_example.sw). This library is intended for convenience and is by no means the only way to implement the standard

I believe the proposed changes make Sway code significantly more readable and less verbose. Having to call `storage.owner.my_function()` for absolutely everything simply doesn't make sense compared to just the function itself. The trade off being well worth the minor inconvenience of a constructor in turn for easier auditability.

File: libs/ownership/README.md [hunk 1]
```diff
@@ -7,61 +7,64 @@
...
-}
-```
-
-2. Leave the owner uninitialized and call the `set_ownership()` function in your own constructor.
-
-```rust
-storage {
-    owner: Ownership = Ownership::uninitialized(),
-}
+Once imported, the Ownership library's functions should be available. To use them, be sure to initialize the owner for your contract by calling the `initialize_ownership()` function in your own constructor.
```
Comment (K1-R1) [inline libs/ownership/README.md:28]:
```suggestion
Once imported, the Ownership library's functions will be available. To use them initialize the owner for your contract by calling the `initialize_ownership()` function in your own constructor method.
```

File: libs/ownership/README.md [hunk 2]
```diff
@@ -7,61 +7,64 @@
...
+
+impl SRC5 for Contract {
+    #[storage(read)]
+    fn owner() -> State {
+        _owner()
+    }
+}
+```
+
+> **NOTE** A constructor must be implemented to initialize the owner.
```
Comment (K1-R1) [inline libs/ownership/README.md:68]:
```suggestion
> **NOTE** A constructor method must be implemented to initialize the owner.
```

**Review by K1-R1 — APPROVED** [review]
Big fan of the UX improvements here. Just a couple of comments for the README, but they're only suggestions

**Review by Braqzen — APPROVED** [review]
<3

**Pull request merged**